CMakeLists.txt

```diff
@@ -25,6 +25,18 @@ endif()
 set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin )
 set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
 
+# Define macro to use atomic implemnetation with GCC built-in.
+# Up until GCC 4.6.3, __sync_* shall be called. From 4.7.0, 
+# __atomic_* are available. We only implemented atomic
+# with __atomic_*.
+#
+# todo: check compiler is GCC AND GCC version is above 4.7.0.
+add_definitions( -DCOMPILER_OPTION_GCC_ATOMIC_BUILTIN=1 )
+
+# Whether to use atomic or not.
+# Default to "no".
+add_definitions( -DIOT_ATOMIC_OPERATION=0 )
```

Comment:
I think it should default to enabled (with compiler support).

Reply:
+1

Reply:
Is the plan to soft launch atomic with this PR? If yes, I'll flip it to 1. 

This macro controls whether libraries depend on atomic. For now, the only place we use atomic in library is MQTT packet identifier, which is more like an example application scenario.